components/oauth-extensions/dynamic-client-manager/src/main/java/org/wso2/carbon/identity/oauth/extension/DynamicClientRegistrationUtil.java

```diff
@@ -228,6 +228,11 @@ public static void unregisterApplication(String userId, String applicationName,
             oAuthAdminService.removeOAuthApplicationData(consumerKey);
 
             ApplicationManagementService appMgtService = ApplicationManagementService.getInstance();
+
+            if (appMgtService == null) {
+                throw new APIManagementException("Error occurred while retrieving Application Management" +
```

Comment:
Shouldnt this be some form of a Runtime Exception? AppMgtService being null indicates to a flaw in making ApplicationManagementService available during the server startup and thus, usually is a bug. So, it might make more sense to have an appropriate Runtime exception thrown in this particular line.